Fix doc for Jenkins on Kubernetes (#7958)
jenkins-infra/jenkins.io: content/doc/book/installing/kubernetes.adoc

Kubernetes' ability to orchestrate container deployment ensures that Jenkins always
has the right amount of resources available.

Hosting Jenkins on a Kubernetes Cluster is beneficial for Kubernetes-based
deployments and dynamic container-based scalable Jenkins agents.

Several strategies for such setup and maintenance are explored below, including:

> * direct use of `kubectl` and YAML files to configure certain aspects of
>   the deployment,
> * or use of `helm` tool and Helm Chart files to manage a whole ecosystem
>   (e.g. both the controller AND a population of agents where actual work
>   happens),
> * or further use of the Jenkins Operator to manage operations for Jenkins
>   on Kubernetes in the cluster.

Reviewer: ```suggestion
* Direct use of `kubectl` and YAML files to configure certain aspects of
  the deployment.
* Use of the `helm` tool and Helm Chart files to manage a whole ecosystem.
  For example, both the controller AND a population of agents where actual work
  happens.
* Further use of the Jenkins Operator to manage operations for Jenkins
  on Kubernetes in the cluster.
```
Small suggestion to make each option its own sentence and adjust syntax.

In this Jenkins Kubernetes deployment we have used the following:

[arabic]

> . The Jenkins container version is referenced in the example YAML above by a floating tag `lts`, so the optional `imagePullPolicy: Always` line is added to pull and deploy a new LTS image release (if exists) *whenever* the pod is restarted. Otherwise, the currently deployed version is used after restart (and there's no other way to tell Kubernetes to pull a newer image, short of locating and deleting the old one from the node's cache).

Reviewer: ```suggestion
. The Jenkins container version is referenced in the example YAML above by a floating tag `lts`, so the optional `imagePullPolicy: Always` line is added to pull and deploy a new LTS image release (if it exists) *whenever* the pod is restarted. Otherwise, the currently deployed version is used after restart (and there's no other way to tell Kubernetes to pull a newer image, short of locating and deleting the old one from the node's cache).
```
Small syntax suggestion.

>   * Note that "whenever the pod is restarted" includes e.g. server reboot or the restart requested in Web-UI to apply downloaded plugin updates.

Reviewer: ```suggestion
  * Note that "whenever the pod is restarted" includes events such as a server reboot or the restart requested in Web-UI to apply downloaded plugin updates.
```
Small syntax suggestion to use "such as" instead of "e.g.".
